File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/Go/s221078931_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)


type IntegralRect struct {
	height int
	width  int
}

func (r IntegralRect) diagonal() int {
	if r.height >= r.width {
		return int(^uint(0) >> 1) // Max int value
	}
	return r.height*r.height + r.width*r.width
}


type ByDiagonal []IntegralRect

func (a ByDiagonal) Len() int           { return len(a) }
func (a ByDiagonal) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
func (a ByDiagonal) Less(i, j int) bool {
	d1 := a[i].diagonal()
	d2 := a[j].diagonal()
	if d1 == d2 {
		return a[i].height < a[j].height
	}
	return d1 < d2
}

func main() {
	var rects [40000]IntegralRect
	for i := 0; i < 200; i++ {
		for j :=  0; j < 200; j++ {
			rects[i*200+j] = IntegralRect{i + 1, j + 1}
		}
	}
	sort.Sort(ByDiagonal(rects[:]))

	scanner := bufio.NewScanner(os.Stdin)
	for scanner.Scan() {
		input := scanner.Text()
		tmpArray := strings.Fields(input)
		h, _ := strconv.Atoi(tmpArray[0])
		w, _ := strconv.Atoi(tmpArray[1])

		if h == 0 && w == 0 {
			break
		}

		tmpRect := IntegralRect{h, w}
		diag := tmpRect.diagonal()

		for i := 0; i < len(rects); i++ {
			if rects[i].diagonal() > diag {
				fmt.Printf("%d %d\n", rects[i].height, rects[i].width)
				break
			}
		}
	}
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/C/s221078931_commented.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_INT 2147483647

typedef struct {
    int height;
    int width;
} IntegralRect;

int diagonal(IntegralRect rect) {
    if (rect.height >= rect.width) {
        return MAX_INT;
    }
    return rect.height * rect.height + rect.width * rect.width;
}

int compare(const void *a, const void *b) {
    IntegralRect *rectA = (IntegralRect *)a;
    IntegralRect *rectB = (IntegralRect *)b;
    int d1 = diagonal(*rectA);
    int d2 = diagonal(*rectB);
    if (d1 == d2) {
        return rectA->height - rectB->height;
    }
    return d1 - d2;
}

int main() {
    IntegralRect rects[40000];
    for (int i = 0; i < 200; i++) {
        for (int j = 0; j < 200; j++) {
            rects[i * 200 + j].height = i + 1;
            rects[i * 200 + j].width = j + 1;
        }
    }
    qsort(rects, 40000, sizeof(IntegralRect), compare);

    while (1) {
        char input[100];
        if (!fgets(input, 100, stdin)) break;
        int h, w;
        if (sscanf(input, "%d %d", &h, &w) != 2) continue;
        if (h == 0 && w == 0) {
            break;
        }
        IntegralRect tmpRect = {h, w};
        int diag = diagonal(tmpRect);
        for (int i = 0; i < 40000; i++) {
            if (compare(&rects[i], &tmpRect) > 0) {
                printf("%d %d\n", rects[i].height, rects[i].width);
                break;
            }
        }
    }
    return 0;
}
// End of Code
